src/main/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -11,152 +12,188 @@
 import java.util.Set;
 
 /**
- * this is an implementation of a Map of JexlVCMatchExp to true or false values.  It lazy initializes each value
- * as requested to save as much processing time as possible.
+ * This is an implementation of a Map of {@link JexlVCMatchExp} to true or false values.
+ * It lazily initializes each value as requested to save as much processing time as possible.
  *
+ * TODO: must resolve the following outdated comment in PR review rounds
  * Compatible with JEXL 1.1 (this code will be easier if we move to 2.0, all of the functionality can go into the
  * JexlContext's get()
  *
+ * TODO: there is some troubling design choices in this class: the laziness and exception behavior:
+ * TODO:   because the class is designed to be lazy, when an instance is created, some invalid expressions may be provided
+ * TODO:   later when user requests some/all values, those invalid expressions will result in IllegalArgumentException,
+ * TODO:   especially when calling the function values(), which is the last thing users expect when calling such accessors.
+ * TODO:   but there may be no good choice here. could decide to "unsupport" the values() function.
  */
 
 class JEXLMap implements Map<JexlVCMatchExp, Boolean> {
-    // our variant context and/or Genotype
-    private final VariantContext vc;
-    private final Genotype g;
 
     // our context
     private JexlContext jContext = null;
+    // our variant context and/or Genotype
```

Comment:
Why move these down like this?  Not a big deal either way but I don't see the reason to change them.